bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Cursor.java

```diff
@@ -120,7 +119,7 @@ public final class Cursor extends Resource {
 public Cursor(Device device, int style) {
 	super(device);
 	this.cursorHandleProvider = new StyleCursorHandleProvider(style);
-	this.handle = this.cursorHandleProvider.createHandle(this, DEFAULT_ZOOM);
+	this.handle = this.cursorHandleProvider.createHandle(device, DEFAULT_ZOOM).getHandle();
```

Comment:
just a Nit: The handle field still stores the raw long, while `zoomLevelToHandle` stores the `CursorHandle`. This  introduces a risk of the two becoming out-of-sync if the logic changes in the future. IMO It might be cleaner to store the `CursorHandle` directly in handle and call `this.handle.getHandle()` wherever the long handle is needed @HeikoKlare

Reply:
It's true that the two values may get out of sync but that's actually not different from the current state. `CursorHandle` is immutable, so it will be as consistent to the current `handle` value as it was before. We need to get rid of the `handle`, but that's something that will be done with the lazy generation of handles implemented by Shahzaib.